_posts/2022-08-29-sbt-tutorial.md

```diff
@@ -0,0 +1,514 @@
+```
+~compile
+```
+Now, sbt will track the source files and if any change is detected in the files, it will automatically recompile them.
+
+## Adding external Dependenncies
+So far, we have created scala files without using any external dependencies. Now, let's look at how we can add additional libraries to our project. We can add the dependencies to out project by adding it to the settings _libraryDependencies_. SBT follows ivy style format to define the dependencies as :
+```
+libraryDependencies += "com.lihaoyi" %% "fansi" % "0.4.0"
+```
+
+The method `+=` appends the provided library to the project dependencies. 
+An sbt dependency contains mainly 3 parts separated by % symbol. 
+The first part is the groupId of the library. The second part is the library name(artifactId) and the third part is the version of the library to be used. If you notice, you can see that a double percentage symbol (%%) is used between groupId and artifactId. 
+Scala is not binary compatible with different versions (such as 2.11, 2.12, 2.13 etc). Hence there are separate releases for each scala libraries for each required versions. %% symbol ensures that sbt uses the same version of library as the project.
+That means, sbt will automatically append the scala version of the project before trying to find the library. The above dependency code is equivalent to the following format(note that single % is used, but artifactid contains the scala major version):
+```
+libraryDependencies += "com.lihaoyi" % "fansi_2.13" % "0.4.0"
+``` 
+
+SBT also has triple % symbol (%%%), which is used to resolve ScalaJS libraries.
+
+We can add multiple dependencies together by using ++= and Seq :
+```
+libraryDependencies ++= Seq(
+  "com.lihaoyi" % "fansi_2.13" % "0.4.0"
+  //Can add more depencies here.
+)
+``` 
+
+## Scala files under project directory
+SBT allows us to keep supporting files for the build as scala files. This helps to configure most of the required configurations in the known format of scala classes. We can place the scala files under the directory _project_. 
+Generally we keep the version info as an object in the scala files and these are accessible directly inside the build.sbt files. This was we can keep and track all the dependencies in a single places and access between multiple parts of the project.
+
+## SBT Console
+SBT also can start a REPL session within the project using the command _sbt console_. This will start up a REPL session just like the scala repl. However, sbt console repl will also load all the dependencies jars to the classpath so that we can directly import the classes and use them. 
+For example, let's try to use the fansi library in the sbt console. 
+We can then start up the sbt session by using _sbt_ command. Then we can use the command _console_ within the sbt session to startup a repl. 
```

Comment:
- _the_  `sbt` command

Reply:
I added one more small part to the sentence (to explicitly mention about `sbt console` and `console` within sbt session